src/test/java/hudson/matrix/MatrixProjectTest.java

```diff
@@ -412,6 +413,24 @@ public void testTwoFileParams() throws Exception {
         j.assertBuildStatusSuccess(f.get(10,TimeUnit.SECONDS));
     }
 
+    @Issue("JENKINS-34758")
+    @Test
+    public void testParametersAsEnvOnChildren() throws Exception {
+        MatrixProject p = createMatrixProject();
+        p.setAxes(new AxisList(new TextAxis("foo","1")));
+        p.addProperty(new ParametersDefinitionProperty(
+            new StringParameterDefinition("MY_PARAM","")
+        ));
+        // must fail if $MY_PARAM or $foo are not defined in children
+        p.getBuildersList().add(new Shell("set -eux; echo $MY_PARAM; echo $foo"));
```

Comment:
Better to use, e.g.

```java
p.getBuildersList().add(new Shell("set -eux; echo got $MY_PARAM and $foo"));
```

and then at the end extract the `MatrixRun` and use `assertLogContains` to look for `got value1 and 1`.